### DISCUSSION TITLE
mqtt_node High memory usage。 version：3.8.14

### DISCUSSION THREAD

**Discussion by feiniao886** [rabbitmq/rabbitmq-server#2866] [Other]
Thank you for using RabbitMQ.

**STOP NOW AND READ THIS** BEFORE OPENING A NEW ISSUE ON GITHUB

Unless you are CERTAIN you have found a reproducible problem in RabbitMQ or
have a **specific, actionable** suggestion for our team, you must first ask
your question or discuss your suspected issue on the mailing list:

https://groups.google.com/forum/#!forum/rabbitmq-users

Team RabbitMQ does not use GitHub issues for discussions, investigations, root
cause analysis and so on.

Please take the time to read the CONTRIBUTING.md document for instructions on
how to effectively ask a question or report a suspected issue:

https://github.com/rabbitmq/rabbitmq-server/blob/master/CONTRIBUTING.md#github-issues

Following these rules **will save time** for both you and RabbitMQ's maintainers.

Thank you.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
There isn't much we can recommend beyond the [tools that provide a breakdown](https://www.rabbitmq.com/memory-use.html) of memory used by a node.

**Comment by michaelklishin** [comment]
This sounds like a duplicate of https://github.com/rabbitmq/rabbitmq-server/discussions/2863 now. Upgrade to `3.8.14` which includes https://github.com/rabbitmq/rabbitmq-server/pull/2692 (well, `3.8.10` does but there is no reason to not use the latest patch release).

**Reply by feiniao886** [reply]
yes
![image](https://user-images.githubusercontent.com/29135686/109947850-bbe38480-7d14-11eb-8e2c-1291eef513bd.png)

**Reply by feiniao886** [reply]
![image](https://user-images.githubusercontent.com/29135686/109949820-cc94fa00-7d16-11eb-82b6-1b1ce6ee7873.png)

**Reply by feiniao886** [reply]
why，

**Reply by kjnilsson** [reply]
Have you got a very high MQTT connection churn?

**Reply by feiniao886** [reply]
The current link has been removed
The current number of connections is 0
mqtt-node Memory usage has been growing.

all queue type is classic.

![image](https://user-images.githubusercontent.com/29135686/109954673-922e5b80-7d1c-11eb-9d86-3dd1702568ee.png)

![image](https://user-images.githubusercontent.com/29135686/109954803-ba1dbf00-7d1c-11eb-9335-e0fa1f69b7e4.png)

![image](https://user-images.githubusercontent.com/29135686/109957550-133b2200-7d20-11eb-9db8-b392467c57fd.png)

**Reply by kjnilsson** [reply]
hm it is still executing the old, inefficient code which means the mqtt machine has not upgraded itself to the current version yet. Is this a clustered environment or a single node RabbitMQ?

**Reply by feiniao886** [reply]
clustered environment 

Did the update succeed?

Such an operation:
systemctl stop rabbitmq-server
yum install rabbitmq-3.8.14xxx.rpm
systemctl start rabbitmq-server

What should I do, reinstall?

![image](https://user-images.githubusercontent.com/29135686/109959079-fb649d80-7d21-11eb-903f-7ea3c6be75e7.png)

**Reply by kjnilsson** [reply]
Are all nodes up to date?

Anyhow you could try decommissioning then disabling the plugin on all nodes as described here https://www.rabbitmq.com/mqtt.html#disabling-plugin

then restart + re-enabling the plugin.

If that fails (it may time out) you could try the nuclear approach of disabling the MQTT plugin, moving all files and folders inside the mnesia/quorum directory on each node to a temporary location, followed by a restart of all nodes then re-enablement of the plugin.

**Reply by feiniao886** [reply]
thank you so much for your suggestion.
After resetting, the problem was resolved.
The current operating status is stable
thx, thx !!!
